### PR TITLE
Parser/tag/mappers

### PR DESCRIPTION
**Pull request by caroo7**
Tag mappers. Retrieving tags when they occured.

### CODE REVIEW

**Comment by sledzgrzegorz** [comment]
I have looked at this code and I think it is ok.

File: parser/src/main/java/converter/BookDetailsToBookAssembler.java [hunk 1]
```diff
@@ -116,15 +117,14 @@ private Book convert(BookDetails bookDetails) {
...
-        String tagString = tagMapper.getMap().get(bookDetails.getTags());
-        Tag tag = tagRepo.findByName(tagString);
-        if(tag != null) {
-            tags.add(tag);
+        if (bookDetails.getTags() == null) {
+            return Collections.emptySet();
         }
-        return tags;
+        return bookDetails.getTags().stream().map(s -> tagRepo.findByName(tagMapper.getMap().get(s)))
+                .filter(tag -> tag != null)
```
Comment (LukaszCzubernat) [inline parser/src/main/java/converter/BookDetailsToBookAssembler.java]: Maybe use Objects.nonNull()

File: parser/src/main/java/converter/BookDetailsToBookAssembler.java [hunk 2]
```diff
@@ -116,15 +117,14 @@ private Book convert(BookDetails bookDetails) {
...
 
-    // todo improve this method - iterate over getTags method
+
     private Set<Tag> retrieveTags(BookDetails bookDetails) {
-        Set<Tag> tags = new HashSet<>();
-        String tagString = tagMapper.getMap().get(bookDetails.getTags());
-        Tag tag = tagRepo.findByName(tagString);
-        if(tag != null) {
-            tags.add(tag);
+        if (bookDetails.getTags() == null) {
```
Comment (LukaszCzubernat) [inline parser/src/main/java/converter/BookDetailsToBookAssembler.java]: Optional?
Or better add Optional to your last 'return' and there filter books without tags

File: parser/src/main/java/converter/BookDetailsToBookAssembler.java [hunk 3]
```diff
@@ -84,8 +85,8 @@ private Book convert(BookDetails bookDetails) {
         if (booksCache.contains(book)) {
             int indexInCache = booksCache.indexOf(book);
             book = booksCache.get(indexInCache);
-        }else {
-            log.debug("Book will be saved on database: "+book);
+        } else {
+            log.debug("Book will be saved on database: " + book);
```
Comment (LukaszCzubernat) [inline parser/src/main/java/converter/BookDetailsToBookAssembler.java:83]: if-else just for logging?

**Comment by LukaszCzubernat** [comment]
ok :+1:

File: cache/src/main/java/access/CacheReader.java [hunk 4]
```diff
@@ -12,10 +12,10 @@
 @Log4j2
 public class CacheReader {
 
-    public List<Book> getBooksFromCache() {
+    public List<Book> getBooksFromCache(String libName) {
 
         List<Book> books = new ArrayList<>();
-        try (FileInputStream fileInputStream = new FileInputStream("cache.txt")) {
+        try (FileInputStream fileInputStream = new FileInputStream(libName)) {
```
Comment (LazyScoundrel) [inline cache/src/main/java/access/CacheReader.java:18]: :+1:

File: database/src/main/java/entities/Library.java [hunk 5]
```diff
@@ -16,23 +17,16 @@
     @GeneratedValue(strategy = GenerationType.IDENTITY)
     private long id;
 
+    @Getter
     private String name;
-
+    @Getter
```
Comment (LazyScoundrel) [inline database/src/main/java/entities/Library.java:22]: :+1:

File: database/src/main/java/entities/Tag.java [hunk 6]
```diff
@@ -11,11 +10,12 @@
 import java.io.Serializable;
 
 @Entity
+@ToString
```
Comment (LazyScoundrel) [inline database/src/main/java/entities/Tag.java]: Do we really want the Lombok's ToString here? You know it's format and it won't be good in terms of viewing

File: database/src/main/java/repositories/BookRepository.java [hunk 7]
```diff
@@ -11,5 +12,6 @@
 
     List<Book> findAll();
 
-}
+    List<Book>  findByLibrary(Library library);
```
Comment (LazyScoundrel) [inline database/src/main/java/repositories/BookRepository.java:15]: :+1:

File: parser/src/main/java/parser/empik/EmpikBookParser.java [hunk 8]
```diff
@@ -23,7 +24,7 @@ public String getTitle(Optional<Document> document) {
         if (document.isPresent()) {
             return  document.get().select("span.pDAuthorList > a").stream().map(element -> element.text()).collect(Collectors.toSet());
         }
-        return null;
+        return Collections.emptySet();
```
Comment (LazyScoundrel) [inline parser/src/main/java/parser/empik/EmpikBookParser.java:27]: :+1:

**Comment by LazyScoundrel** [comment]
Overall these are good changes :)

**Pull request merged**